File: src/content/slicesOverlap.ts
import { StringSlice } from './types';

/**
 * Returns whether the given slices overlap.
 */
const slicesOverlap = ([start1, end1]: StringSlice, [start2, end2]: StringSlice) =>
  !(end1 < start1 || end2 < start2 || end1 <= start2 || end2 <= start1)

export default slicesOverlap;

File: npbloom-core/src/commonMain/kotlin/content/SlicesOverlap.kt
package content

/**
 * Returns whether the given slices overlap.
 */
internal fun slicesOverlap(slice1: StringSlice, slice2: StringSlice): Boolean {
    val (start1, end1) = slice1
    val (start2, end2) = slice2
    return !(end1 < start1 || end2 < start2 || end1 <= start2 || end2 <= start1)
}
